### PR TITLE
[MISC] Move core/pod_tuple.hpp -> utility/tuple/pod_tuple.hpp

### PR DESCRIPTION
**Pull request by eaasna**
Part of [seqan/product_backlog#160](https://github.com/seqan/product_backlog/issues/160)

- Moves core/pod_tuple.hpp -> utility/tuple/pod_tuple.hpp
- Updates ingroup to utility
- Deprecates core/pod_tuple.hpp
- Fixes include paths
- Moves pod_tuple_test.hpp and pod_tuple_snippet according to new directory structure
- Adjusts CMakeLists.txt files

### CODE REVIEW

File: include/seqan3/core/all.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 #include <seqan3/core/debug_stream.hpp>
 #include <seqan3/core/detail/all.hpp>
 #include <seqan3/core/platform.hpp>
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/utility/tuple/pod_tuple.hpp>
```
Comment (MitraDarja) [inline include/seqan3/core/all.hpp:27]: Alphabetical order?

File: include/seqan3/core/tuple_utility.hpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 #include <utility>
 
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/utility/tuple/pod_tuple.hpp>
 #include <seqan3/utility/tuple/concept.hpp>
```
Comment (MitraDarja) [inline include/seqan3/core/tuple_utility.hpp:18]: Alphabetical order

File: test/snippet/utility/tuple/pod_tuple.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/utility/tuple/pod_tuple.hpp>
 #include <seqan3/core/debug_stream.hpp>
```
Comment (MitraDarja) [inline test/snippet/utility/tuple/pod_tuple.cpp:2]: Alphabetical order

**Review by MitraDarja — CHANGES_REQUESTED** [review]
Looks good, just one minor thing, headers are ordered alphabetically. If the order is at the moment incorrect, but will be correct once another file is moved from core to utility I think you can leave the incorrect order (I didn't mark this), all other of my marks should be reordered if I am not mistaken. :)

File: include/seqan3/core/pod_tuple.hpp [hunk 4]
```diff
@@ -6,371 +6,12 @@
...
- */
-template <template <typename ...> typename t, typename ...types>
-    requires std::is_base_of_v<seqan3::pod_tuple<types...>, t<types...>>
-struct tuple_size<t<types...>> :
-    public std::integral_constant<std::size_t, sizeof...(types)>
-{};
-
-} // namespace std
+SEQAN3_DEPRECATED_HEADER(
+"This header is deprecated and will be removed in SeqAn-3.1.0 Please #include <seqan3/utility/tuple/pod_tuple.hpp> instead.")
```
Comment (eseiler) [inline include/seqan3/core/pod_tuple.hpp:17]: You are missing a new line here (files always end with a new line). All editor usually have a setting for this somewhere :)

Comment (eaasna) [reply include/seqan3/core/pod_tuple.hpp]: Fixed it! Thanks for noticing :)

File: test/snippet/utility/tuple/pod_tuple.cpp (hunk 3 above)
Comment (eaasna) [reply test/snippet/utility/tuple/pod_tuple.cpp]: Fixed the include order in the files you mentioned. Please review again :)

**Comment by eaasna** [comment]
I fixed the merge conflicts and rebased my branch so this should now be good to go :) 

I also went back and cleaned up the commit history. The seqan contributor guide said to keep the number of commits minimal, wondering if it's wise to have just one bigger commit in this case?

**Review by MitraDarja — APPROVED** [review]
LGTM. Thanks. 👍

File: test/unit/utility/tuple_utility_test.cpp [hunk 5]
```diff
@@ -5,19 +5,16 @@
 // shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 // -----------------------------------------------------------------------------------------------------
 
-#include <seqan3/std/concepts>
-#include <tuple>
-#include <type_traits>
-
```
Comment (marehr) [inline test/unit/utility/tuple_utility_test.cpp:11]:
```suggestion
#include <seqan3/std/concepts>
#include <tuple>
#include <type_traits>

```

File: test/unit/utility/tuple_utility_test.cpp [hunk 6]
```diff
@@ -5,19 +5,16 @@
...
 #include <gtest/gtest.h>
 
 #include <seqan3/core/detail/strong_type.hpp>
-#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/std/concepts>
+#include <tuple>
+#include <type_traits>
```
Comment (marehr) [inline test/unit/utility/tuple_utility_test.cpp:13]:
```suggestion
```

**Review by marehr — COMMENTED** [review]
Thank you!

**Review by marehr — APPROVED** [review]
I committed the suggested changes for you! 

Thank you for the PR!

I approve now, and please ping someone (if I forget) to merge this as soon as the CI is done.

**Comment by eaasna** [comment]
> I committed the suggested changes for you!
> 
> Thank you for the PR!
> 
> I approve now, and please ping someone (if I forget) to merge this as soon as the CI is done.

Thanks, but what is CI? @marehr

**Comment by smehringer** [comment]
> > I committed the suggested changes for you!
> > Thank you for the PR!
> > I approve now, and please ping someone (if I forget) to merge this as soon as the CI is done.
> 
> Thanks, but what is CI? @marehr

CI is the abbreviation for *Continous Integration*, referring to the tests/checks that are running in every PR.

**Pull request merged**